bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromHandler.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    private static final int HTTP_OK_CODE = 200;
+    private static final String COMMUNICATION_ERROR = "Error while communicating to the myStrom plug: ";
+    private static final String HTTP_REQUEST_URL_PREFIX = "http://";
+    private static final String HTTPS_REQUEST_URL_PREFIX = "https://";
+
+    private final Logger logger = LoggerFactory.getLogger(MyStromHandler.class);
+
+    private HttpClient httpClient;
+    private String hostname = "";
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private ExpiringCache<MyStromReport> cache = new ExpiringCache<>(Duration.ofSeconds(3), this::getReport);
+    private final Gson gson = new Gson();
+
+    public MyStromHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                pollDevice();
+            } else {
+                if (command instanceof OnOffType && CHANNEL_SWITCH.equals(channelUID.getId())) {
+                    sendHttpGet("relay?state=" + (command == OnOffType.ON ? "1" : "0"));
+                    scheduler.schedule(this::pollDevice, 500, TimeUnit.MILLISECONDS);
+                }
+            }
+        } catch (MyStromException e) {
+            logger.warn("Error while handling command {}", e.getMessage());
+        }
+    }
+
+    private @Nullable MyStromReport getReport() {
+        try {
+            String returnContent = sendHttpGet("report");
+            MyStromReport report = gson.fromJson(returnContent, MyStromReport.class);
+            updateStatus(ThingStatus.ONLINE);
+            return report;
+        } catch (MyStromException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return null;
+        }
+    }
+
+    private void pollDevice() {
+        MyStromReport report = cache.getValue();
+        if (report != null) {
+            updateState(CHANNEL_SWITCH, report.relay ? OnOffType.ON : OnOffType.OFF);
+            updateState(CHANNEL_POWER, QuantityType.valueOf(report.power, WATT));
+            updateState(CHANNEL_TEMPERATURE, QuantityType.valueOf(report.temperature, CELSIUS));
+        }
+    }
+
+    @Override
+    public void initialize() {
+        MyStromConfiguration config = getConfigAs(MyStromConfiguration.class);
+        if (config.hostname.startsWith(HTTP_REQUEST_URL_PREFIX)
+                || config.hostname.startsWith(HTTPS_REQUEST_URL_PREFIX)) {
+            this.hostname = config.hostname;
+        } else {
+            this.hostname = HTTP_REQUEST_URL_PREFIX + config.hostname;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollDevice, 0, config.refresh, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
```

Comment:
Base class destructors should be called at the end of the implementing destructors.

Reply:
done